src/diff/props.js

```diff
@@ -39,39 +38,28 @@ const XLINK_NS = 'http://www.w3.org/1999/xlink';
  * @param {boolean} isSvg Whether or not this DOM node is an SVG node or not
  */
 function setProperty(dom, name, value, oldValue, isSvg) {
+	let v;
+
 	name = isSvg ? (name==='className' ? 'class' : name) : (name==='class' ? 'className' : name);
 
 	if (name==='style') {
+		// Always clear the previous styles
+		dom.style.cssText = EMPTY_OBJ;
 
-		/* Possible golfing activities for setting styles:
-		 *   - we could just drop String style values. They're not supported in other VDOM libs.
-		 *   - assigning to .style sets .style.cssText - TODO: benchmark this, might not be worth the bytes.
-		 *   - assigning also casts to String, and ignores invalid values. This means assigning an Object clears all styles.
-		 */
-		let s = dom.style;
+		for (let i in value) {
+			v = value[i];
+			v = typeof v==='number' && IS_NON_DIMENSIONAL.test(i)===false ? (v + 'px') : v;
 
-		if (typeof value==='string') {
-			s.cssText = value;
-		}
-		else {
-			if (typeof oldValue==='string') s.cssText = '';
-			else {
-				// remove values not in the new list
-				for (let i in oldValue) {
-					if (value==null || !(i in value)) s.setProperty(i.replace(CAMEL_REG, '-'), '');
-				}
+			// For css vars, just define them with `setProperty`;
+			if (/^--/.test(i)) {
+				dom.style.setProperty(i, v);
 			}
-			for (let i in value) {
-				const v = value[i];
-				if (oldValue==null || v!==oldValue[i]) {
-					s.setProperty(i.replace(CAMEL_REG, '-'), typeof v==='number' && IS_NON_DIMENSIONAL.test(i)===false ? (v + 'px') : v);
-				}
+			else {
+				// Bench for setter vs (setProperty + regex) https://esbench.com/bench/5ccb5a284cd7e6009ef6232e
+				dom.style[i] = v;
```

Comment:
These are almost the same in terms of perf. The upside is in code size.

Reply:
I think like this we support IE9 again.

Reply:
Not anymore 😦

Reply:
The `dom.style[i]` thing is a clever way of getting rid of the camel case transformation, and something like this seems to get rid of 6 extra bytes:

```js
const v = (value && (i in value))
	? (typeof set[i]==='number' && IS_NON_DIMENSIONAL.test(i)===false)
		? set[i] + 'px'
		: set[i]
	: '';
if (i[0] === '-' && i[1] === '-') {
	dom.style.setProperty(i, v);
}
else {
	dom.style[i] = v;
}
```

Reply:
The problem with assignment is that is a setter that would trigger parsing for that property. `setProperty` is way less overhead, hence the need to do the camelCase to dash-case parsing. Check-out this bench from @developit 
![image](https://user-images.githubusercontent.com/3405161/57359789-34ad4000-7181-11e9-98f2-41db689082f8.png)
https://esbench.com/bench/5730613ddb965b9a00965a5b

Reply:
Ah, thats true. Running the benchmark on Chrome 74 + MacOS 10.14 gives the following results:
<img width="633" alt="Screenshot 2019-05-08 at 11 47 02" src="https://user-images.githubusercontent.com/19776768/57363070-e8fe9480-7188-11e9-8c69-dfc4655eba1d.png">
It must be noted that this benchmark doesn't take into account for the `setProperty` approach to run the property names through the regex. With small modifications to the `setProperty` approach that more closely resemble this pull request (run property names through the regex + use `setProperty(a, '')` instead of `removeProperty(a)`) the difference becomes somewhat smaller:
<img width="553" alt="Screenshot 2019-05-08 at 11 58 10" src="https://user-images.githubusercontent.com/19776768/57363191-3975f200-7189-11e9-8063-67fb87bf4102.png">
Interestingly on Safari 12.1 and Firefox 66 the `properties with check` approach is faster (after the mentioned modifications to the benchmark).
Safari:
<img width="468" alt="Screenshot 2019-05-08 at 11 56 08" src="https://user-images.githubusercontent.com/19776768/57363348-970a3e80-7189-11e9-9b88-74782ab6e6c2.png">
Firefox:
<img width="578" alt="Screenshot 2019-05-08 at 11 52 24" src="https://user-images.githubusercontent.com/19776768/57363366-a5f0f100-7189-11e9-89c0-1c8b0915d9b8.png">

Reply:
Seems on par with my initial bench, tbh https://esbench.com/bench/5ccb5a284cd7e6009ef6232e.

@developit so, does it make sense? doing `setProperty` + regex is in the same category as calling the `setter`? By doing just the `setter` we're shaving some nice `6B` 😄 . I'm gonna try to create some more randomly, uncache-able code path and see if that still holds.